plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -342,6 +342,12 @@ public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggrega
             return Optional.empty();
         }
         PinotTableHandle tableHandle = (PinotTableHandle) handle;
+
+        List<PinotColumnHandle> groupingColumns = getOnlyElement(groupingSets).stream()
+                .map(PinotColumnHandle.class::cast)
+                .map(PinotMetadata::toNonAggregateColumnHandle)
+                .collect(toImmutableList());
```

Comment:
This change on its own is not good as it is better to do things when they are needed. Here it looks like we can exit from the method without using `groupingColumns` variable. I would suggest to squash that change with some other change that would justify it.